### PR TITLE
chat: version subscriptions

### PR DESCRIPTION
**Pull request by liam-fitzgerald**
Implements new version negotiation logic

### CODE REVIEW

File: desk/app/chat.hoon [hunk 1]
```diff
@@ -41,10 +44,14 @@
       abet:init:cor
     [cards this]
   ::
-  ++  on-save  !>(state)
+  ++  on-save  !>(state(old-epic our-epic))
   ++  on-load
     |=  =vase
     ^-  (quip card _this)
+    ?:  =(1 2)
```
Comment (arthyn) [inline desk/app/chat.hoon:51]: Is this just a guard until we enable this?

**Review by arthyn — APPROVED** [review]
lgtm assuming I understood this correctly. we store what epic we're on and also subscribe to new epics from the host. when an epic changes, we take that fact and store it so that we're on the correct subscription?

File: desk/app/chat.hoon (hunk 1 above)
Comment (liam-fitzgerald) [reply desk/app/chat.hoon:51]: yes

File: desk/lib/tale.hoon [hunk 2]
```diff
@@ -0,0 +1,14 @@
+/-  g=groups
```
[the remark this answers is not in the document]
Comment (liam-fitzgerald) [reply desk/lib/tale.hoon:1]: whoops

[the remark this answers is not in the document]
Comment (liam-fitzgerald) [reply desk/lib/tale.hoon:1]: whoops

**Comment by liam-fitzgerald** [comment]
- every new subscription begins with an %epic as the first fact
- when an application upgrades, send the new epic if it changed on every versioned subscription
- if the foreign epic matches our epic, then proceed as normal (there's no versioning issues)
- if the foreign epic is less than our epic (we received an OTA that the publisher does not have), subscribe to the foreign epic and restart the subscription when it resynchronises
- if the foreign epic is greater than our epic (publisher received an OTA that we don't have), store the foreign epic, then in the +on-load, check for epics that we resynchronised with

**Comment by arthyn** [comment]
good to merge?

**Pull request merged**